src/components/BookList.js

```diff
@@ -1,37 +1,65 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
+import axios from 'axios';
 import Book from './Book';
 import BookForm from './BookForm';
-import { removed } from '../Redux/books/booksSlice';
+import { getBooks } from '../Redux/books/booksSlice';
 import BookCategory from './BookCategory';
 import ChapterCounter from './ChapterCounter';
 
 const getRandomChapter = () => Math.floor(Math.random() * 30) + 1;
 
 const BookList = () => {
-  const books = useSelector((state) => state.books);
+  const { loading, books, error } = useSelector((state) => state.books);
   const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getBooks());
+  }, [dispatch]);
 
-  const handleRemoveBook = (bookId) => {
-    dispatch(removed(bookId));
+  const handleRemoveBook = (key) => {
+    const url = `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/IfJm8IrVNYBvy5mLAJXh/books/${key}`;
+
+    axios.delete(url)
+      .then((response) => {
+        console.log('Response:', response.data);
+        dispatch(getBooks());
+      })
+      .catch((error) => {
+        console.error('Failed to delete book:', error);
```

Comment:
> - [ ] Please Could you please eliminate the usage of console.log and console.error from your code? These statements are contributing to the untidiness of your code. 😞
> 
> Also, I suggest you place the Axios DELETE request directly in the redux bookSlice, as it follows the recommended pattern for handling asynchronous operations with Redux. Keeping API calls and state management logic separate is a good practice. With createAsyncThunk, you encapsulate the API call logic in the Redux action creator rather than scattering it throughout your components.😊

```
export const removeBook = createAsyncThunk(
  'books/removeBook',
  async (bookId) => {
    try {
      await axios.delete(`url/${bookId}`);
      return bookId;
    } catch (err) {
      throw err;
    }
  },
);
```